docs/guide/dev/integration/react/README.md

```diff
@@ -215,6 +169,206 @@ component.editor.getData();
+
+```js
+const editor = CKEDITOR.instances.myeditor;
+```
+
+The editor element is also accessible via the [component](#editor-instance) itself.
+
+### `readOnly`
+
+`Boolean`
+
+Sets the {@link features/readonly/README read-only mode}:
+
+```jsx
+<CKEditor
+	data="<p>Editor's content</p>"
+	readOnly={true}
+/>
+```
+
+This property takes precedence over the {@linkapi CKEDITOR.config#readOnly `config.readOnly`} setting.
+
+Defaults to `false`.
+
+### `style`
+
+`Object`
+
+Style rules set that will be applied to {@linkapi CKEDITOR.editor#container editor container} with {@linkapi CKEDITOR.dom.element#setStyles}.
+
+```jsx
+<CKEditor
+	data="<p>Editor's content</p>"
+	style={{
+		'margin-top': '100px',
+		'border': '5px solid red'
+	}}
+/>
+```
+
+Requires component [type](#type) property to be `classic`.
+
+### `type`
+
+`'classic'` | `'inline'`
+
+By default, the CKEditor 4 React component creates {@link guide/dev/framed/README classic editor}. To create an {@link guide/dev/inline/README inline editor}, add the `type` property with the value of `inline` to the `<CKEditor />` tag:
+
+```jsx
+<CKEditor
+	data="<p>Some initial data</p>"
+	type="inline"
+/>
+```
+
+You can also explicitly set the `type` property to `classic` to create the classic editor:
+
+```jsx
+<CKEditor
+	data="<p>Some initial data</p>"
+	type="classic"
+/>
+```
+
+Every other value of the `type` property will be treated as `classic`.
+
+For more details, look at the {@link guide/dev/ckeditor_js_load/README Loading CKEditor Script article}.
+
+Defaults to `'classic'`.
+
+### `onBeforeLoad`
+
+`Function`
+
+Callback function with single argument: `CKEDITOR` namespace. It is invoked **each time new editor instance is loaded**, but `CKEDITOR` object always refers to the same namespace.
+
+**Note**: To modify `CKEDITOR` namespace it is recommended to use [onNamespaceLoaded](#onnamespaceloaded) event.
+
+Simple usage:
+
+```jsx
+<CKEditor
+	data="<p>First editor.</p>"
+	onBeforeLoad={ CKEDITOR => {
+		console.log( 'First editor loaded!' );
+	}
+/>
+
+<CKEditor
+	data="<p>Second editor.</p>"
+	onBeforeLoad={ CKEDITOR => {
+		console.log( 'Second editor loaded!' );
+	}
+/>
+```
+
+An example below shows incorrect usage - it attempts to load the same plugin two times to the same namespace:
+
+```jsx
+<CKEditor
+	name="editorOne"
+	data="<p>Editor's content</p>"
+	onBeforeLoad={ CKEDITOR => {
+			// Add external `placeholder` plugin which will be available for each
+			// editor instance on the page.
+			CKEDITOR.plugins.addExternal( 'placeholder', '/path/to/the/placeholder/plugin', 'plugin.js' );
+		}
+	}
+/>
+
+<CKEditor
+	name="editorTwo"
+	data="<p>Editor's content</p>"
+	onBeforeLoad={ CKEDITOR => {
+			// Namespace was already loaded, but this callback will be called anyway.
+			// Unnecessary adding the same plugin to the namespace.
+			CKEDITOR.plugins.addExternal( 'placeholder', '/path/to/the/placeholder/plugin', 'plugin.js' );
+		}
+	}
+/>
+```
```

Comment:
I'm not sure about giving incorrect usage example. Some people skim docs looking for code samples only so may wrongly assume it is correct usage. OTOH the code comments explains what's going on :thinking: WDYT @sculpt0r @Dumluregn?

Reply:
Maybe we could use the warning box styling here to be sure we catch user's attention?

Reply:
> Some people skim docs looking for code samples only so may wrongly assume it is correct usage.

> Maybe we could use the warning box styling here to be sure we catch user's attention?

Agree :)

Reply:
:+1: